pkg/utils/pod.go

```diff
@@ -54,3 +55,18 @@ func GetPodCondition(status *v1.PodStatus, conditionType v1.PodConditionType) (i
 	}
 	return -1, nil
 }
+
+// CalculatePodRequests sum request total from pods
+func CalculatePodRequests(pods []v1.Pod, resource v1.ResourceName) (int64, error) {
+	var requests int64
+	for _, pod := range pods {
+		for _, c := range pod.Spec.Containers {
+			if containerRequest, ok := c.Resources.Requests[resource]; ok {
+				requests += containerRequest.MilliValue()
+			} else {
+				return 0, fmt.Errorf("missing request for %s", resource)
+			}
+		}
+	}
+	return requests, nil
+}
```

Comment:
This function ignore the init container, what's the consideration of this?

Reply:
Autoscaling is dealing with running pod & containers, init container is not a long running one that should not count in here.